### ISSUE TITLE
Should not print lint errors when formatting stdin

### ISSUE THREAD

**Issue by tjmehta** [Flet/standard-engine#22]
### Example

fibonacci.js
```js
module.exports = fibonacci

function fibonacci (n) {
  var a = 0, b = 1, f = 1;
  for (var i = 2; i <= n; i++) {
      f = a + b;
      a = b;
      b = f;
  }
  return f;
}
```

Run standard format w/ stdin option on file:
```bash
cat fibonacci.js | standard --format --stdin
```

Prints:
```js
module.exports = fibonacci

function fibonacci (n) {
  var a = 0, b = 1, f = 1
  for (var i = 2; i <= n; i++) {
    f = a + b
    a = b
    b = f
  }
  return f
}
standard: Use JavaScript Standard Style (https://github.com/feross/standard)
standard:   <text>:4:3: Split initialized 'var' declarations into multiple statements.
```

I would expect to be able to do the following:

```bash
cat fibonacci.js | standard --format --stdin > fibonacci-formatted.js
```

However, this would result in a invalid javascript file (bc the lint warnings would be appended to the bottom of the file)

**Comment by Flet** [comment]
I think redirection just `stdout` to a file *should* be the right answer here:
```
cat fibonacci.js | standard --format --stdin 1> fibonacci-formatted.js
```

but it looks like the first banner line is sent via `stout` not `stderr`, so it is appended to the bottom:
```
🕛 ❯ cat fibonacci-formatted.js 
module.exports = fibonacci

function fibonacci (n) {
  var a = 0, b = 1, f = 1
  for (var i = 2; i <= n; i++) {
    f = a + b
    a = b
    b = f
  }
  return f
}
standard: Use JavaScript Standard Style (https://github.com/feross/standard) 
```

So, what if we updated the banner to output to `stdout` instead? That would allow just stout to be redirected via `1>`... Would this work for you?

I like the idea of the `--format` flag to still lint as it makes the command automatically fix what it can and still throw linting errors. This can be useful if folks have it as part of their test script.

Another way to solve this would be to use the `standard-format` command directly.

@feross any thoughts?

**Comment by tjmehta** [comment]
I discovered this problem while making a patch to [sublime-standard-format](https://github.com/bcomnes/sublime-standard-format).

It appears that your proposed solution would not work with the way that this plugin is implemented.
I don't know python well enough to know if this is common practice. If this is a common pattern in python (and maybe other languages) I would suggest not outputting the lint errors to `stderr`.

// Links to implementation of sublime-standard-format:
https://github.com/bcomnes/sublime-standard-format/blob/c33988ebc4a7662adc9d977c28b00127f2e147d1/standard-format.py#L98
https://github.com/bcomnes/sublime-standard-format/blob/c33988ebc4a7662adc9d977c28b00127f2e147d1/standard-format.py#L142
// python docs for `Popen.communicate()`
https://docs.python.org/2/library/subprocess.html?highlight=communicate#subprocess.Popen.communicate

**Comment by tjmehta** [comment]
thoughts?

**Comment by Flet** [comment]
@bcomnes thoughts?

**Comment by feross** [comment]
@Flet 

> So, what if we updated the banner to output to stdout instead? 

I think you meant to say that we should output the initial banner to `stderr`, not to `stdout`. That's the solution here, IMO. I'll send a PR.

**Comment by feross** [comment]
PR out: https://github.com/Flet/standard-engine/pull/42

**Comment by Flet** [comment]
Thanks @feross!

Merged and published as `standard-engine@2.2.5`

**Comment by dcousens** [comment]
:+1:
